components/elm/src/biogeochem/AllocationMod.F90

```diff
@@ -1547,6 +1532,23 @@ subroutine Allocation2_ResolveNPLimit (bounds, num_soilc, filter_soilc  , &
                      end if
                   end do
 
+                  ! potential adsorption rate without plant and microbial interaction
+                  ! including weathering, deposition, phosphatase, mineralization, immobilization, plant uptake
+                  dsolutionp_dt(c,j) = gross_pmin_vr(c,j) -potential_immob_p_vr(c,j) - &
+                       col_plant_pdemand_vr(c,j) + biochem_pmin_vr_col(c,j) + &
+                       primp_to_labilep_vr_col(c,j) + pdep_to_sminp(c) *ndep_prof(c,j)
+                  adsorb_to_labilep_vr(c,j) = (vmax_minsurf_p_vr(isoilorder(c),j)* km_minsurf_p_vr(isoilorder(c),j)) / &
+                       ((km_minsurf_p_vr(isoilorder(c),j)+max(solutionp_vr(c,j),0._r8))**2._r8 ) * dsolutionp_dt(c,j)
+                  ! sign convention: if adsorb_to_labilep_vr(c,j) < 0, then it's desorption
+                  if (adsorb_to_labilep_vr(c,j) >= 0.0_r8) then
+                     adsorb_to_labilep_vr(c,j) = max(min(adsorb_to_labilep_vr(c,j), &
+                          (vmax_minsurf_p_vr(isoilorder(c),j) - labilep_vr(c,j))/dt),0.0_r8)
+                     desorb_to_solutionp_vr(c,j) = 0.0_r8
+                  else
+                     desorb_to_solutionp_vr(c,j) = min(-1.0*adsorb_to_labilep_vr(c,j), labilep_vr(c,j)/dt)
+                     adsorb_to_labilep_vr(c,j) = 0.0_r8
+                  end if
+
```

Comment:
@qzhu-lbl @rgknox It appears this change is already in master. Correct?

Reply:
yes, should be already in